apps/condo/domains/common/components/Card/CardsContainer.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { Col, Row } from 'antd'
+
+export type CardsPerRowType = 1 | 2 | 3 | 4 | 5
+
+interface CardsContainerProps {
+    cardsPerRow?: CardsPerRowType
+    autosize?: boolean
+    minCardWidth?: number
+}
+
+const MIN_CARD_WIDTH = 300
+const MAX_CARDS_PER_ROW = 20
+
+
+// todo Use it in miniapps too
+export const CardsContainer: React.FC<CardsContainerProps> = ({ cardsPerRow = 3, autosize, minCardWidth = MIN_CARD_WIDTH, children }) => {
+    const [colSpan, setColSpan] = useState<number>(24 / cardsPerRow)
+    const containerRef = useRef<HTMLDivElement>()
+
+    useEffect(() => {
+        if (!autosize) return
+
+        const element = containerRef.current
+        const resizeObserver = new ResizeObserver((entries) => {
```

Comment:
Is this made for adaptive cards? Couldn't it be easier?

Reply:
This is the best way to make cards responsive. The size of the container depends not only on the size of the window, but also on the state of the sidebar (open or closed).